pages/home.jsx

```diff
@@ -146,11 +149,121 @@ var BlogSection = React.createClass({
   }
 });
 
+var validateSignupForm = function(signUpFormState) {
+  var errors = [];
+  // regex copied from http://stackoverflow.com/a/46181
+  var regexEmail = /^([\w-]+(?:\.[\w-]+)*)@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$/i;
+  if (!regexEmail.test(signUpFormState.email)) {
+    errors.push("Please enter an email address.");
+  }
+
+  return errors;
+};
+
+var ModalPledge = React.createClass({
+  mixins: [React.addons.LinkedStateMixin],
+  getInitialState: function() {
+    return {
+      email: "",
+      validationErrors: []
+    };
+  },
+  handleSubmit: function(e) {
+    var validationErrors = validateSignupForm(_.pick(this.state,"email"));
+
+    if (validationErrors.length) {
+      e.preventDefault();
+      this.setState({validationErrors: validationErrors});
+      return;
+    }
+
+    if (process.env.NODE_ENV !== 'production' &&
+        !process.env.PLEDGE_MAILINGLIST_URL) {
+      e.preventDefault();
+      alert("PLEDGE_MAILINGLIST_URL is not defined. Simulating " +
+            "a successful pledge signup now.");
+      window.location = "?pledge=thanks";
+    }
+  },
+  renderValidationErrors: function() {
+    if (this.state.validationErrors.length) {
+      return (
+        <div className="alert alert-danger">
+          <p className="error-msg">Please enter an email address.</p>
```

Comment:
> If we don't anticipate more validation errors, we could just have `state.validationErrors` be a boolean, maybe even rename it to `state.invalidEmail` so that the code is clearer... if not we should probably do the whole `.map()` thing to show all the errors.

I'm hoping to turn that fake privacy policy checkbox into a standard one.  See [my comment](https://github.com/mozilla/teach.webmaker.org/pull/1039/files#r32792452).  After we have decided what to do with it, I'll clean up this chunk of code.

And yea, I thought about refactoring the code when I copied the form validation code from `events.jsx`.  But I *felt* this pledge CTA is more or less temporary that's why I got lazy about the factoring work... :smiling_imp: :grimacing:   If we do decide to factor out a `ValidationErrors` React component, do we want to land the changes in this PR or a separate one?